modules/javafx.graphics/src/main/java/javafx/scene/CssStyleHelper.java

```diff
@@ -571,6 +571,140 @@ boolean pseudoClassStateChanged(PseudoClass pseudoClass) {
+            if (inheritOnly && cssMetaData.isInherits() == false) {
+                continue;
+            }
+
+            // Skip the lookup if we know there isn't a chance for this property
+            // to be set (usually due to a "bind").
+            if (!cssMetaData.isSettable(node)) {
+                continue;
+            }
+
+            final String property = cssMetaData.getProperty();
+            boolean isFontProperty = property.equals("-fx-font") || property.equals("-fx-font-size");
+            // This method is executed as a result of change in font size of the control, hence font property should be skipped.
+            if (isFontProperty) {
+                continue;
+            }
+
+            CascadingStyle style = getStyle(node, property, styleMap, transitionStates[0]);
+            if (style != null) {
+                final ParsedValue cssValue = style.getParsedValue();
+                ObjectProperty<StyleOrigin> whence = new SimpleObjectProperty<>(style.getOrigin());
+                ParsedValue resolved = resolveLookups(node, cssValue, styleMap, transitionStates[0], whence, new HashSet<>());
+                boolean isRelative = ParsedValueImpl.containsFontRelativeSize(resolved, false);
+                if (!isRelative) {
+                    continue;
+                }
+            } else {
+                final List<CssMetaData<? extends Styleable, ?>> subProperties = cssMetaData.getSubProperties();
+                final int numSubProperties = (subProperties != null) ? subProperties.size() : 0;
+                if (numSubProperties == 0) {
+                    continue;
+                } else {
+                    // TODO: further optimization
+                    // determine a way to find if any of the sub properties are specified with relative size.
+                    // if none is relatively sized then continue;
+                }
+            }
+
+            // If code flow reaches here then it means that the property is
+            // explicitly specified in css style by user with a relative size.
+            // We should not use the cached value as relative
+            // sized properties must be recalculated when font size changes.
+
+            CalculatedValue calculatedValue = lookup(node, cssMetaData, styleMap, transitionStates[0],
+                        node, cachedFont);
+
+            // lookup is not supposed to return null.
+            if (calculatedValue == null || calculatedValue == SKIP) {
+                continue;
+            }
+
+            try {
+                StyleableProperty styleableProperty = cssMetaData.getStyleableProperty(node);
+                final StyleOrigin originOfCurrentValue = styleableProperty.getStyleOrigin();
+                final StyleOrigin originOfCalculatedValue = calculatedValue.getOrigin();
+
+                if (originOfCalculatedValue == null) {
+                    assert false : styleableProperty.toString();
```

Comment:
We don't generally use `assert` in our code (although I see this part is copied from the original method, so I understand if you prefer to leave it).

Reply:
Removed assert.